hummingbot/core/rate_oracle/rate_oracle.py

```diff
@@ -183,10 +183,20 @@ async def rate_async(self, pair: str) -> Decimal:
         prices = await self._source.get_prices(quote_token=self._quote_token)
         return find_rate(prices, pair)
 
+    def set_price(self, pair: str, price: Decimal):
+        """
+        Update keys in self._prices with new prices
+        """
+        self._prices[pair] = price
+
     async def _fetch_price_loop(self):
         while True:
             try:
-                self._prices = await self._source.get_prices(quote_token=self._quote_token)
+                new_prices = await self._source.get_prices(quote_token=self._quote_token)
```

Comment:
this is going to decrease the performance of the rate oracle since it usually queries a lot of tokens and updating them with a for loop is probably a slow option. I would recommend updating the dictionary directly

Reply:
Thanks, changed from looping to using `.update()` method